plugins/@grouparoo/clickhouse/src/lib/export/exportRecord.ts

```
@@ -0,0 +1,141 @@
+        );
+
+        if (columnsToErase.length > 0) {
+          const nullData = {};
+          columnsToErase.forEach((k) => (nullData[k] = null));
+
+          await connection.asyncQuery(`ALTER TABLE ?? UPDATE ? WHERE ?? = ?`, [
+            table,
+            nullData,
+            primaryKey,
+            newRecordProperties[primaryKey],
+          ]);
+        }
+      } else {
+        // delete & insert
+        await connection.asyncQuery(`ALTER TABLE ?? DELETE WHERE ?? = ?`, [
+          table,
+          primaryKey,
+          newRecordProperties[primaryKey],
+        ]);
+
+        await connection.asyncQuery(`INSERT INTO ?? (??) VALUES (?)`, [
+          table,
+          Object.keys(newRecordProperties),
+          Object.values(newRecordProperties),
+        ]);
+      }
+    } else {
+      // just insert
+      await connection.asyncQuery(`INSERT INTO ?? (??) VALUES (?)`, [
+        table,
+        Object.keys(newRecordProperties),
+        Object.values(newRecordProperties),
+      ]);
+    }
+
+    // --- Groups --- //
+
+    await connection.asyncQuery(`ALTER TABLE ?? DELETE WHERE ?? = ?`, [
+      groupsTable,
+      groupForeignKey,
+      newRecordProperties[primaryKey],
+    ]);
+
+    if (toDelete) {
+      return;
+    }
+
+    for (const i in newGroups) {
+      const data = {
+        [groupForeignKey]: newRecordProperties[primaryKey],
+        [groupColumnName]: newGroups[i],
+      };
+
+      const query = `INSERT INTO ?? (??) SELECT ? WHERE NOT EXISTS (SELECT ?? FROM ?? WHERE ?? = ? AND ?? = ?)`;
+      const params = [
+        groupsTable,
+        Object.keys(data),
+        Object.values(data),
+        groupForeignKey,
+        groupsTable,
+        groupForeignKey,
+        data[groupForeignKey],
+        groupColumnName,
+        data[groupColumnName],
+      ];
+
+      await connection.asyncQuery(query, params);
+    }
+  } catch (e) {
+    error = e;
+  } finally {
+    if (error) {
+      throw error;
+    }
```

Why not just let this throw? Or is `error` getting grabbed at a higher level?

---

I'm not sure but this is how it's implemented in all the SQL-based plugins.